### ISSUE TITLE
Error in stoploss on exchange (Huobi)

### ISSUE THREAD

**Issue by STraderMX** [freqtrade/freqtrade#8003]
## Describe your environment

  * Operating system: linux (Debian 11)
  * Python Version: 3.9.2
  * CCXT version: 2.4.60
  * Freqtrade Version: 2022.12
  
## Describe the problem:

Stoploss on exchange is getting errors on Huobi, I must say that same config and strategy works perfectly on binance and kucoin.  

This is the current config (private info and whitelist pairs arent' included):
```json
{
  "max_open_trades": 4,
  "stake_currency": "USDT",
  "stake_amount": "unlimited",
  "tradable_balance_ratio": 0.99,
  "fiat_display_currency": "USD",
  "dry_run": false,
  "dry_run_wallet": 1000,
  "cancel_open_orders_on_exit": false,
  "unfilledtimeout": {
    "entry": 10,
    "exit": 10,
    "exit_timeout_count": 0,
    "unit": "minutes"
  },
  "entry_pricing": {
    "price_side": "other",
    "use_order_book": false,
    "order_book_top": 1,
    "price_last_balance": 0.0,
    "check_depth_of_market": {
      "enabled": false,
      "bids_to_ask_delta": 1
    }
  },
  "exit_pricing": {
    "price_side": "other",
    "use_order_book": false,
    "order_book_top": 1
  },
  "order_types": {
    "entry": "market",
    "exit": "market",
    "emergency_exit": "market",
    "force_exit": "market",
    "force_entry": "market",
    "stoploss": "limit",
    "stoploss_on_exchange": true,
    "stoploss_on_exchange_interval": 300,
    "stoploss_on_exchange_limit_ratio": 0.99
  },
  "exchange": {
    "ccxt_config": { "enableRateLimit": true },
    "ccxt_async_config": {
      "enableRateLimit": true,
      "rateLimit": 200
    }
  },
  "pairlists": [{ "method": "StaticPairList" }],
  "edge": {
    "enabled": false,
    "process_throttle_secs": 3600,
    "calculate_since_number_of_days": 7,
    "allowed_risk": 0.01,
    "stoploss_range_min": -0.01,
    "stoploss_range_max": -0.1,
    "stoploss_range_step": -0.01,
    "minimum_winrate": 0.6,
    "minimum_expectancy": 0.2,
    "min_trade_number": 10,
    "max_trade_duration_minute": 1440,
    "remove_pumps": false
  },
  "initial_state": "running",
  "force_entry_enable": false,
  "internals": {
    "process_throttle_secs": 300,
    "heartbeat_interval": 60
  },
  "dataformat_ohlcv": "hdf5",
  "dataformat_trades": "hdf5"
}
```

And this is the log error in in the bot:
```python
2023-01-09 19:40:04 - freqtrade.exchange.exchange - INFO - stoploss limit order added for MASK/USDT. stop price: 2.665019074034461. limit: 2.6384
2023-01-09 19:45:01 - freqtrade.worker - INFO - Bot heartbeat. PID=624, version='2022.12, strategy_version: 1.0', state='RUNNING'
2023-01-09 19:45:04 - freqtrade.freqtradebot - INFO - Cancelling current stoploss on exchange for pair MASK/USDT (orderid:715855203243149) in order to add another one ...
2023-01-09 19:45:04 - freqtrade.exchange.common - WARNING - stoploss() returned exception: "Could not place stoploss order due to ExchangeError. Message: huobi {"status":"error","err-code":"order-stop-order-hit-trigger","err-msg":"Orders that are triggered immediately are not supported.","data":null}". Giving up.
2023-01-09 19:45:04 - freqtrade.freqtradebot - ERROR - Unable to place a stoploss order on exchange.
Traceback (most recent call last):
  File "/root/freqtrade/freqtrade/exchange/exchange.py", line 1174, in stoploss
    order = self._api.create_order(symbol=pair, type=ordertype, side=side,
  File "/root/freqtrade/.env/lib/python3.9/site-packages/ccxt/huobi.py", line 3914, in create_order
    return getattr(self, method)(symbol, type, side, amount, price, query)
  File "/root/freqtrade/.env/lib/python3.9/site-packages/ccxt/huobi.py", line 3985, in create_spot_order
    response = self.spotPrivatePostV1OrderOrdersPlace(self.extend(request, params))
  File "/root/freqtrade/.env/lib/python3.9/site-packages/ccxt/base/exchange.py", line 502, in inner
    return entry(_self, **inner_kwargs)
  File "/root/freqtrade/.env/lib/python3.9/site-packages/ccxt/base/exchange.py", line 2803, in request
    return self.fetch2(path, api, method, params, headers, body, config, context)
  File "/root/freqtrade/.env/lib/python3.9/site-packages/ccxt/base/exchange.py", line 2800, in fetch2
    return self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File "/root/freqtrade/.env/lib/python3.9/site-packages/ccxt/base/exchange.py", line 664, in fetch
    self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File "/root/freqtrade/.env/lib/python3.9/site-packages/ccxt/huobi.py", line 5369, in handle_errors
    raise ExchangeError(feedback)
ccxt.base.errors.ExchangeError: huobi {"status":"error","err-code":"order-stop-order-hit-trigger","err-msg":"Orders that are triggered immediately are not supported.","data":null}

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/root/freqtrade/freqtrade/freqtradebot.py", line 1088, in create_stoploss_order
    stoploss_order = self.exchange.stoploss(
  File "/root/freqtrade/freqtrade/exchange/common.py", line 179, in wrapper
    raise ex
  File "/root/freqtrade/freqtrade/exchange/common.py", line 164, in wrapper
    return f(*args, **kwargs)
  File "/root/freqtrade/freqtrade/exchange/exchange.py", line 1196, in stoploss
    raise TemporaryError(
freqtrade.exceptions.TemporaryError: Could not place stoploss order due to ExchangeError. Message: huobi {"status":"error","err-code":"order-stop-order-hit-trigger","err-msg":"Orders that are triggered immediately are not supported.","data":null}
2023-01-09 19:45:04 - freqtrade.freqtradebot - WARNING - Could not create trailing stoploss order for pair MASK/USDT.
```

And so on this error is repeated every time the bot tries to set the stoploss, in fact I can see the history tries in the exchange:
<img width="1914" alt="image" src="https://user-images.githubusercontent.com/122330178/211446076-874f251b-b42a-45d8-95f9-3ae7ca879957.png">

For now I stoped it, if you need further info please let me know.

Thanks!

**Comment by xmatthias** [comment]
the log on the exchange clearly shows that some stoplosses go through - which means that technically - the stoploss code works, but suggests that your stoploss settings are pretty tight.

What's your stoploss (or rather, trailing stoploss) settings (should be from the strategy)?

**Comment by STraderMX** [comment]
> the log on the exchange clearly shows that some stoplosses go through - which means that technically - the stoploss code works, but suggests that your stoploss settings are pretty tight.
> 
> What's your stoploss (or rather, trailing stoploss) settings (should be from the strategy)?

The stoploss is 0.001, I'm using the same strategy on binance and kucoin without this error, in fact, yesterday after noticing this error there was a open trade in positive profit, then I tried a forcesell from telegram and just received the "exiting unrealized Profit" but never the never the "exited" message.

**Comment by xmatthias** [comment]
a 0.1% stoploss is way too low for quickly moving crypto markets - that's basically like setting a "sell now" order - and is therefore not supported (not by the exchange - and freqtrade will raise this error).

The error is exactly what it says - your stop would trigger immediately. 
Binance would have exactly the same error (given the same market movements) - and kucoin most likely too (though i don't know the exact wording there)..

it'll obviously depend on the market movements - but we've seen this problem a lot in the past - and it's always been with too tight stoplosses for the market you're in.

The [stoploss on exchange](https://www.freqtrade.io/en/stable/stoploss/#stop-loss-on-exchangefreqtrade) documentation explicitly mentions this in the "Note" box:

> Do not set too low/tight stoploss value if using stop loss on exchange!
> If set to low/tight then you have greater risk of missing fill on the order and stoploss will not work.

0.1% is clearly too tight - with most markets, 0.5% SHOULD work - though on some low-priced markets, that can also be too low (look at price movements - if it moves more than your stoploss in a given second - your stop is too tight).

**Comment by STraderMX** [comment]
I checked the logs of binance and kucoin, in binance I found a couple of logs similar, but no raising errors, just the warning.

**Comment by xmatthias** [comment]
ah now i see what you mean - the error comming out of ccxt is wrong (should be invalidOrder).

will be fixed with https://github.com/ccxt/ccxt/pull/16425.

**Comment by STraderMX** [comment]
> ah now i see what you mean - the error comming out of ccxt is wrong (should be invalidOrder).
> 
> will be fixed with [ccxt/ccxt#16425](https://github.com/ccxt/ccxt/pull/16425).

Exactly, thanks!

**Comment by xmatthias** [comment]
If you're not on docker, you can safely update ccxt with `pip install -U ccxt` and use that for huobi (shouldn't cause any issues).

if you're using binance futures though, please do NOT use the new ccxt version at this point - it'll not be functional without updates to the freqtrade version, too - which means i currently can't bump the version in the repository.
